### PR TITLE
Run tests as project admin

### PR DESCRIPTION
**Pull request by mgencur**
## Description

This is a follow up to https://github.com/knative/client/pull/1383 (this PR depends on it) and fixes the remaining sub-issues from https://github.com/knative/client/issues/1304
As a result, it will be possible to run the test suite as a project admin (vs. cluster-admin).

## Changes

* Create Broker explicitly rather than by labeling namespace
* Use Role instead of ClusterRole to work with Sources
* Separate SourceList test for cluster admin and regular user
* Instructions for running E2E as project admin

## Reference

Fixes https://github.com/knative/client/issues/1352

### CODE REVIEW

File: test/e2e/source_list_crd_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or im
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+// +build !serving
+// +build !project_admin
```
Comment (rhuss) [inline test/e2e/source_list_crd_test.go:17]: I would include some comment to the test, that this test only works for cluster admin and essentially the comment that you added to the issue. That way people are more aware that this test is special. 

We should also open an issue to fix this in the kn code and add a link to the issue here.

**Review by rhuss — COMMENTED** [review]
Thanks @mgencur !

Looks good to me with a minor comment to add a comment ;-)

**Comment by mgencur** [comment]
Thanks @rhuss , pushed.

File: test/e2e/source_list_crd_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	"gotest.tools/v3/assert"
+
+	"knative.dev/client/lib/test"
+	"knative.dev/client/pkg/util"
+)
+
+// This test requires cluster admin permissions due to working with CustomResourceDefinitions.
+// It can be excluded from test execution by using the project_admin build tag.
+// See https://github.com/knative/client/issues/1385
```
Comment (rhuss) [inline test/e2e/source_list_crd_test.go:32]: 👍

**Review by rhuss — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**